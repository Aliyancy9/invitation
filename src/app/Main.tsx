import * as React from 'react'
import { makeStyles } from '@material-ui/styles'
import {AppBar, Typography, Button } from '@material-ui/core'
import EmailDialog from './EmailDialog'

const useStyles = makeStyles({
    container: {
        height:'100%',
        backgroundColor: 'white'
    },
    header: {
        height: '5rem',
    },
    headerText: {
        color: 'grey',
        display: 'flex',
        alignItems: 'center',
        height: '5rem',
        marginLeft: '3rem',
        fontSize: '2rem',
    }, 
    footer: {
        height: '5rem',
        top: 'auto',
        bottom: 0,
      },
    
    footerText: {
        color: 'grey',
        height: '5rem',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        flexDirection: 'column',
        fontStyle: 'oblique',
    }, 
    content: {
        height: '100%',
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        flexDirection: 'column',
    },
    text: {
        margin: '1rem 0',
    },
    button: {
        margin: '1rem 0',
    },

})

const Main: React.FC = () => {
    const classes = useStyles({})

    const [open, setOpen] = React.useState(false)

    const onToggle = () => setOpen(!open)

return (
    <div className={classes.container}>
        <AppBar
            className={classes.header}
            color = 'inherit'
        >
            <Typography className={classes.headerText}>
                 Broccoli & Co.
            </Typography>
        </AppBar>
        <div className={classes.content}>
            
            <Typography variant = 'h3' color = 'textSecondary'>
                A better way
            </Typography>
            <Typography variant = 'h3' color = 'textSecondary'>
                to enjoy every day.
            </Typography>
            
            <Typography variant = 'h6' color = 'textSecondary' className={classes.text}>
                Be the first to know when we launch.
            </Typography>

            <Button 
                variant="outlined" 
                className={classes.button}
                onClick={onToggle}
            >
                Request an invite
            </Button>
        </div>
            <EmailDialog 
                isOpen={open}
                onClose={onToggle}
            />
        <AppBar
            className={classes.footer}
            color = 'inherit'
        >
            <div className={classes.footerText}>
                <Typography >
                Made with &hearts; in Melbourne.  
                </Typography>
                <Typography>
                Â© 2016 Broccoli & Co. All rights reserved.   
                </Typography>
            </div>
        </AppBar>
    </div>
)}

export default Main