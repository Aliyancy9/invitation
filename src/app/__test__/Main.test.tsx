import * as React from 'react'
import { render } from '@testing-library/react'

import Main from '../Main'

const renderTest = (comp: any) => render(comp)

describe('EmailDialog', () => {
    it('should show correct title', () => {   
    const { container } = renderTest(<Main  />)
    expect(container).toMatchSnapshot()
    })
})