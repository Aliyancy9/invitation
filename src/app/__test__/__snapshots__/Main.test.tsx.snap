// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailDialog should show correct title 1`] = `
<div>
  <div
    class="makeStyles-container-1"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorInherit makeStyles-header-2 mui-fixed MuiPaper-elevation4"
    >
      <p
        class="MuiTypography-root makeStyles-headerText-3 MuiTypography-body1"
      >
        Broccoli & Co.
      </p>
    </header>
    <div
      class="makeStyles-content-6"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-colorTextSecondary"
      >
        A better way
      </h3>
      <h3
        class="MuiTypography-root MuiTypography-h3 MuiTypography-colorTextSecondary"
      >
        to enjoy every day.
      </h3>
      <h6
        class="MuiTypography-root makeStyles-text-7 MuiTypography-h6 MuiTypography-colorTextSecondary"
      >
        Be the first to know when we launch.
      </h6>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-8"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Request an invite
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorInherit makeStyles-footer-4 mui-fixed MuiPaper-elevation4"
    >
      <div
        class="makeStyles-footerText-5"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Made with ♥ in Melbourne.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          © 2016 Broccoli & Co. All rights reserved.
        </p>
      </div>
    </header>
  </div>
</div>
`;
